src/requestor/routes/maintain.py

```diff
@@ -203,30 +208,33 @@ async def update_request(
 
 @router.delete("/request/{request_id}", status_code=HTTP_200_OK)
 async def delete_request(
-    api_request: Request,
     request_id: uuid.UUID,
-    bearer_token: HTTPAuthorizationCredentials = Security(bearer),
+    auth=Depends(Auth),
 ) -> dict:
     """
     Delete an access request.
     """
-    request = await RequestModel.query.where(
-        RequestModel.request_id == request_id
-    ).gino.first_or_404()
+    logger.debug(f"Deleting request '{request_id}'")
+    async with db.transaction():
+        request = (
+            await RequestModel.delete.where(RequestModel.request_id == request_id)
+            .returning(*RequestModel)
+            .gino.first_or_404()
+        )
 
-    await authorize(
-        api_request.app.arborist_client,
-        bearer_token,
-        "delete",
-        [request.to_dict()["resource_path"]],
-    )
+        authorized = await auth.authorize(
+            "delete",
+            [request.resource_path],
+            throw=False,
+        )
+
+        if not authorized:
+            # will rollback the transaction
+            raise HTTPException(
+                HTTP_403_FORBIDDEN,
+                "Permission denied",
+            )
```

Comment:
@fantix you mentioned we should do [this](https://python-gino.org/docs/en/master/reference/api/gino.transaction.html#gino.transaction.GinoTransaction.raise_rollback) `tx.raise_rollback()`

but the following doesn't work - it doesn't reach the `HTTPException`
```
if not authorized:
  tx.raise_rollback()
  raise HTTPException(
    HTTP_403_FORBIDDEN,
    "Permission denied",
  )
```

Reply:
Wondering if this comment is still relevant 🤔 I don't know Gino well enough. Will we automatically roll back if an exception was raised?

Reply:
i think it's still relevant. i don't know Gino that well either but this unit test should be enough to make sure the code behaves the right way: https://github.com/uc-cdis/requestor/blob/cecf37d173b4a2a789cd2b144348fe55df80d722/tests/test_maintain.py#L293

Reply:
👍